ttn/application.py

```diff
@@ -31,32 +31,28 @@
 class ApplicationClient:
 
     def __init__(self, app_id, token_or_key,
-                 net_address="", certificate="",
+                 net_address="", cert_content="",
                  discovery_address="discovery.thethings.network:1900",
                  path_to_key=""):
         self.app_id = app_id
-        self.discovery_address = discovery_address
-        self.net_address = net_address
-        self.credentials = certificate
-        self.path_to_key = path_to_key
 
-        if not self.path_to_key:
+        if not path_to_key:
             self.app_access_key = token_or_key
-        elif is_token(token_or_key, self.path_to_key):
+        elif is_token(token_or_key, path_to_key):
             self.app_access_token = token_or_key
         else:
             raise RuntimeError("The key or token provided is invalid")
 
-        if not self.net_address:
-            discovery = DiscoveryClient(self.discovery_address)
+        if not net_address:
+            discovery = DiscoveryClient(discovery_address)
             announcement = discovery.get_by_app_id(self.app_id)
-            self.net_address = announcement.net_address
-            self.credentials = announcement.certificate
-        elif not hasattr(self, "credentials"):
+            net_address = announcement.net_address
+            cert_content = announcement.certificate
+        elif not cert_content:
```

Comment:
Why do you need to provide a `cert_content` if you provide a `net_address`?

Reply:
It's passed to gRPC to create a secure channel, so I thought it could be a problem if it's an empty string but maybe not

Reply:
In case you're not sure how TLS works: gRPC uses (**optionally**) TLS for encryption, just like how HTTPS works under the hood (in modern browsers, at least). This means that the security of a connection is guaranteed by TLS certificates - that a certificate is valid, your OS verifies that it is cryptographically signed by a Certificate Authority (e.g. Comodo, GoDaddy, and such). It is also possible to create your own Certificate Authority by generating and installing root certificates, but in that case, you'll only be recognised as a Certificate Authority on the machines you install the root certificate on. To know more about how TLS works, you can have a look at those videos ([1](https://www.youtube.com/watch?v=SJJmoDZ3il8), [2](https://www.youtube.com/watch?v=iQsKdtjwtYI)) or at those articles ([1](https://medium.com/eleven-labs/understanding-ssl-tls-c140b3a08532), [2](https://medium.com/@User3141592/what-is-ssl-and-how-does-it-work-a5465d19b494)).

What this means in our case is that certificates are optional for all network servers that have certificate signed by certificate authorities (the public community networks, private networks). It's only required when you want to connect over TLS to a network server with a self-signed certificate. Let's remove the certificate requirement - there might be cases where you want to connect without TLS (e.g. over a local connection), or where you want to connect over TLS but that the certificate is already signed by a certificate authority (e.g. TTI networks).